### ISSUE TITLE
`groups.json` uses dash, command JSON uses underscore

### ISSUE THREAD

**Issue by stockholmux** [valkey-io/valkey-doc#178]
I'm working on valkey-io/valkey-io.github.io#150 using the [`groups.json`](https://github.com/valkey-io/valkey-doc/blob/main/groups.json) file to get human readable descriptions from command JSON `group` fields.

Everything corresponds between the files except sorted set in command JSON files are grouped as [`sorted_set`](https://github.com/valkey-io/valkey/blob/unstable/src/commands/zadd.json#L5) and the `groups.json` uses `sorted-set`.

I can work around this but it's annoying that we're not consistent. I'd be in favour of changing `groups.json` to reflect the field in the command JSON. Will this break the man page generation script?

**Comment by zuiderkwast** [comment]
> I'd be in favour of changing groups.json to reflect the field in the command JSON. Will this break the man page generation script?

No, I would just change the man page stuff at the same time.

Will it break the website?

**Comment by stockholmux** [comment]
@zuiderkwast Nope. I would have a bit of [dead code](https://github.com/valkey-io/valkey-io.github.io/blob/736f8425e2ef17ab25d708a0da70e0a6b637a4a4/templates/commands.html#L37) to transform the underscores to dashes but it's fine.

**Comment by stockholmux** [comment]
I'll twitch every time I see the line that transforms the `-` to the `_` but it's not worth a breaking change.

What a maddeningly inconsistent thing. Oh well.

**Comment by stockholmux** [comment]
Although we should probably document that changing this file `valkey-doc` could make a breaking change in `valkey-server` 😮

**Comment by zuiderkwast** [comment]
> I'll twitch every time I see the line that transforms the `-` to the `_` but it's not worth a breaking change.

I'll show you the code that turns `sorted-set` into `sorted_set` for COMMAND DOCS. Beware, it may get you into an infinite-twitch-loop, so please hold someone's hand. It's C code generated by Python code. An array of names that must be in the same order as another array with other names in another file in a different language, namely `server.h`...

* https://github.com/valkey-io/valkey/blob/8.0.1/utils/generate-command-code.py#L581
* https://github.com/valkey-io/valkey/blob/8.0.1/src/server.h#L2396

> Although we should probably document that changing this file `valkey-doc` could make a breaking change in `valkey-server` 😮

No, it doesn't. The code doesn't depend on the doc repo, so we can change the doc repo without considering it a breaking change. That won't solve the inconsistency though, as long as the name `sorted-set` exists in the output of COMMAND DOCS, which might be (arguably?) the recommended way to get this metadata rather than reading the command JSON files directly. I just mean that if we want to solve the inconsistency completely, including in COMMAND DOCS, that would be a breaking change.

Anyway, I don't really think it's worth changing `sorted_sets` to `sorted-sets` in groups.json at this point. We could have used COMMAND DOCS instead of the command JSON files and then we would have hade `sorted-sets` everywhere.
